src/java/main/org/apache/zookeeper/server/quorum/QuorumPeer.java

```diff
@@ -198,6 +246,12 @@ public void recreateSocketAddresses() {
         public long id;
         
         public LearnerType type = LearnerType.PARTICIPANT;
+
+        /**
+         * the time, in milliseconds, before {@link InetAddress#isReachable} aborts
+         * in {@link #getReachableAddress}.
+         */
+        private int ipReachableTimeout = 0;
```

Comment:
I think we can remove this. This will also remove the four copies of exact same initialization code fragment that you use to initialize the value. We can get the sys property and parse the value inside recreateSocketAddress directly.